src/python/seccomp.pyx

```diff
@@ -343,6 +351,189 @@ cdef class Arg:
         """
         return self._arg
 
+cdef class Notification:
+    """ Python object representing a seccomp notification.
+    """
+    cdef libseccomp.seccomp_notif _data
+
+    def __cinit__(self, id, pid, flags, syscall, ip, args):
+        """ Initialize the notification.
+
+        Arguments:
+        id - the notification ID
+        pid - the process ID
+        flags - the notification flags
+        syscall - the syscall number
+        ip - the instruction pointer
+        args - list of the six syscall arguments
+
+        Description:
+        Create a seccomp Notification object.
+        """
+        self._data.id = id
+        self._data.pid = pid
+        self._data.flags = flags
+        self._data.data.nr = syscall
+        self._data.data.instruction_pointer = ip
+        self._data.data.args[0] = args[0]
+        self._data.data.args[1] = args[1]
+        self._data.data.args[2] = args[2]
+        self._data.data.args[3] = args[3]
+        self._data.data.args[4] = args[4]
+        self._data.data.args[5] = args[5]
+
+    def get_id(self):
```

Comment:
I need to preface my comments here by saying I am *DEFINITELY* not Python expert; to be brutally honest, I can barely write Python.

That said, the property decorator seems like an awfully heavyweight solution to this problem.  The Notification object returns a standard Python list via the get_syscall_args() method so users of the Python bindings should be able to interact with the Notification object using a "pythonic" interface; this really only affects the binding implementation and isn't a general problem.

FWIW, the reason we have to do these assignments in the first place is that Cython isn't able to convert between a uint64_t array and a Python list; other array types do work (e.g. unsigned long), but since the args structure is a kernel struct I'm not sure I want to mess around to much with the types.

Reply:
Sounds reasonable to me.  I haven't run the following, but I think it would provide a lighter-weight (if I am understanding correctly) `@property` solution:

```
   @property
   def id(self):
       return self._data.id

    # no need to have an id.setter

...

   @property
   def syscall_args(self):
       args = [self._data.data.args[0],
                self._data.data.args[1],
                self._data.data.args[2],
                self._data.data.args[3],
                self._data.data.args[4],
                self._data.data.args[5]]
        return args
```

Reply:
Then a user like `respond_notify()` can access it via `resp.id = response.id`

Reply:
Hmm, okay, I think I misunderstood both your original comment and Python properties.  Let me take a crack at this ...

Reply:
I've had good luck with `@property` in the past.  But I admit that I haven't used it with Cython, though I don't see why that would matter.

Reply:
Okay, check out the latest push, I believe it has the all property stuff done correctly.  I *think*.